src/validation.cpp

```diff
@@ -2636,6 +2635,7 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
     blockundo.vtxundo.reserve(block.vtx.size() - 1);
     for (unsigned int i = 0; i < block.vtx.size(); i++)
     {
+        if (!state.IsValid()) break;
```

Comment:
just to check: this is just a belt and suspenders? It seems each call site to `state.Invalid` in the loop is followed by a `break`.

Reply:
the state could already be invalid before entering the loop (BIP30 check a few lines above).

Reply:
glossed over that above change for bip30 checks, thanks